src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -72,10 +68,80 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOExceptio
         attributes.put("X-Amz-Expires", "30");
     }
 
+    DescribeInstances(AwsConfig awsConfig) {
+        this.awsConfig = awsConfig;
+    }
+
+    void checkKeysFromIamRoles(Environment env) throws IOException {
+        if (awsConfig.getAccessKey() != null) {
+            return;
+        }
+
+        if (awsConfig.getIamRole() == null) {

```

Comment:
I think for the following conditions, we should search for default iam-role credentials first.
* non-existence of `<iam-role>` tag
* `<iam-role/>` with no role name
* `<iam-role>DEFAULT</iam-role>` 

If we fail then we should fall back to ECS based `iamrole` search which seems to be done by `getKeysFromIamTaskRole(env);`

if that fails as well then we should throw `InvalidConfigurationException` with proper message.

so it would be like that

if (awsConfig.getIamRole() != null) {
getKeysFromIamRole()
}

if (awsConfig.getIamRole() == null || awsConfig.getIamRole().equals("") || awsConfig.getIamRole().equals("DEFAULT")  {
tryGetDefaultIamRole();
}

getKeysFromIamTaskRole(env); // return if that works.

`InvalidConfigurationException` if none of above works.